check-format.sh

```diff
@@ -9,17 +9,34 @@ if [ "$#" -eq 0 ]; then
   exit 1
 fi
 
+fix=false
+while getopts ":f:" opt; do
+  case $opt in
+    f)
+      fix=true
+      shift
+    ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+      exit
+    ;;
+  esac
+done
+
 echo "Checking file format in" "$@"
 
 unformatted_files=""
-for f in $(find "$@" -name "*.h" -or -name "*.hpp" -or -name "*.cpp" -or -name "*.c"); do
+for file in $(find "$@" -name "*.h" -or -name "*.hpp" -or -name "*.cpp" -or -name "*.c"); do
   # Workaround for https://bugs.llvm.org/show_bug.cgi?id=39216
-  d=$(cat "$f" | clang-format-7 -style=file --assume-filename "${f%.*}".cpp | diff "$f" -)
+  d=$(cat "$file" | clang-format-7 -style=file --assume-filename "${file%.*}".cpp | diff "$file" -)
+  if $fix ; then
+    cat "$file" | clang-format-7 -style=file --assume-filename "${file%.*}".cpp > "$file".tmp
```

Comment:
I think clang-format should be able to format the files inplace, rather than via a temporary file

```
  -i                        - Inplace edit <file>s, if specified.
```

Reply:
cool let me check

Reply:
we can't use -i when we are using stdin, and we are using stdin to workaround this known bug: https://bugs.llvm.org/show_bug.cgi?id=39216 as mentioned in the script comment

Reply:
That suffers from the same Objective-C misdetection problem as the check does though.